roles/k3s/setup/tasks/main.yml

```diff
@@ -11,18 +11,33 @@
   with_items:
     - "127.0.0.1 {{ inventory_hostname }}"
 
-- name: Ensure K3s node service is disabled
+- name: Ensure K3s master service is disabled
   systemd:
-    name: k3s-node
+    name: k3s
     daemon_reload: yes
-    state: stopped
     enabled: no
+  when: not master_service
```

Comment:
a slightly more elegant way is to use instead:
```
    enabled: {{ k3s_node_service_enabled }}
```
where *master.yml* is:
```
---
k3s_master_service_status: yes
k3s_node_service_status: no
```

That way you don't need to have tasks to disable or enable the service.
Just consume the enable value and that'll set the service status properly.